### PR TITLE
Set buffer size io.copy

### PR DESCRIPTION
**Pull request by willcAND**
### What

As a result of the performance testing on staging, io.Copy was throwing a timeout error.

We have increased the read & write timeouts and also added a buffer to the copy method.

### How to review

Inspect code, confirm approach.

To test this properly, JMeter must be configured in same the way it is for the performance tests on staging.

### Who can review

A member of the ONS.

### CODE REVIEW

File: service/initialise.go [hunk 1]
```diff
@@ -49,6 +50,8 @@ func (e *ExternalServiceList) GetRequestMiddleware() RequestMiddleware {
 // DoGetHTTPServer creates an HTTP Server with the provided bind address and router
 func (e *Init) DoGetHTTPServer(bindAddr string, router http.Handler) HTTPServer {
 	s := dphttp.NewServer(bindAddr, router)
+	s.Server.ReadTimeout = 15 * time.Second
+	s.Server.WriteTimeout = 30 * time.Second
 	s.HandleOSSignals = false
```
Comment (lindenmckenzie) [inline service/initialise.go]: how sure are we on these? Would we want these to be config vars?

**Review by lindenmckenzie — COMMENTED** [review]
Minor comment.

File: service/initialise.go (hunk 1 above)
Comment (willcAND) [reply service/initialise.go]: We've run JMeter tests with these value as their default (read 5s, write 10s) and the error found on the logs from performance testing is appearing locally. With this increase, the error is no longer present locally. Just going to lower them again and find the point where the errors start to persist.

Comment (willcAND) [reply service/initialise.go]: Also happy to set them in the config

Comment (lindenmckenzie) [reply service/initialise.go]: seems like if we're not sure, then altering via config seems a sensible approach.

File: service/service.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceLi
 
 	// The proxy needs to be set up after the HealthCheck route has been added to the router: in the Setup method, the
 	// proxy adds a catch-all route, so any other routes added after that one will never be reachable.
-	p := proxy.Setup(ctx, r, cfg)
+	p := proxy.Setup(r, cfg)
```
Comment (gedge) [inline service/service.go]: as per my comment on Slack, I would not remove the general context

File: service/initialise.go (hunk 1 above)
Comment (willcAND) [reply service/initialise.go]: Have added these values to the config

**Review by gedge — APPROVED** [review]

**Pull request merged**